src/clj/game/core/access.clj

```diff
@@ -563,7 +566,9 @@
                     (and pos-max?
                          pos-total-cards?
                          only-card)
-                    (access-card state side eid only-card)
+                    (if (= (second (get-zone only-card)) (first server))
+                      (access-card state side eid only-card)
+                      (effect-completed state side eid))
```

Comment:
Are these and the similar lines necessary?

Reply:
Yeah - if the only card you're allowed to access isn't actually on the server you're breaching (ie ash on HQ, beatriz on HQ to breach R&D, losing the ash trace), then we shouldn't be accessing that card (the ash) - since it's not already a candidate for this access.

Reply:
wait, yes they are necessary, since this function is picking the actual card to access

Reply:
Okay, I see the point now.